packages/@aws-cdk/core/lib/type-hints.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Type hints for resolved values
+ */
+export enum ResolutionTypeHint {
+  /**
+   * adsf
+   */
+  STRING = 'string',
+  /**
+   * adsf
+   */
+  NUMBER = 'number',
+  /**
+   * adsf
+   */
+  LIST = 'list',
+  /**
+   * adsf
+   */
+  ERROR = 'error',
```

Comment:
only used in this class:

```/**
 * This marker token is used by PhysicalName.GENERATE_IF_NEEDED. When that token is passed to the
 * physicalName property of a Resource, it triggers different behavior in the Resource constructor
 * that will allow emission of a generated physical name (when the resource is used across
 * environments) or undefined (when the resource is not shared).
 *
 * This token throws an Error when it is resolved, as a way to prevent inadvertent mis-uses of it.
 */
export class GeneratedWhenNeededMarker implements IResolvable {
  public readonly creationStack: string[] = [];
  public readonly typeHint: ResolutionTypeHint;

  constructor() {
    Object.defineProperty(this, GENERATE_IF_NEEDED_SYMBOL, { value: true });
    this.typeHint = ResolutionTypeHint.ERROR;
  }

  public resolve(_ctx: IResolveContext): never {
    throw new Error('Invalid physical name passed to CloudFormation. Use "this.physicalName" instead');
  }

  public toString(): string {
    return 'PhysicalName.GENERATE_IF_NEEDED';
  }
}```

since resolving it will throw an error, we should never even get to the point of reading this type hint...but if we somehow did, better to error out than try to use it as a string.

Reply:
I only added this since I originally had `typeHint` as a required property...it probably makes more sense to just remove this entirely for now than to have extra logic.